pandas/io/stata.py

```diff
@@ -3114,19 +3164,121 @@ def _convert_strls(self, data):
         ]
 
         if convert_cols:
-            ssw = StataStrLWriter(data, convert_cols)
+            ssw = StataStrLWriter(data, convert_cols, version=self._dta_version)
             tab, new_data = ssw.generate_table()
             data = new_data
             self._strl_blob = ssw.generate_blob(tab)
         return data
 
-    def _set_formats_and_types(self, data, dtypes):
+    def _set_formats_and_types(self, dtypes):
         self.typlist = []
         self.fmtlist = []
         for col, dtype in dtypes.items():
             force_strl = col in self._convert_strl
             fmt = _dtype_to_default_stata_fmt(
-                dtype, data[col], dta_version=117, force_strl=force_strl
+                dtype,
+                self.data[col],
+                dta_version=self._dta_version,
+                force_strl=force_strl,
             )
             self.fmtlist.append(fmt)
-            self.typlist.append(_dtype_to_stata_type_117(dtype, data[col], force_strl))
+            self.typlist.append(
+                _dtype_to_stata_type_117(dtype, self.data[col], force_strl)
+            )
+
+
+class StataWriter118(StataWriter117):
+    """
+    A class for writing Stata binary dta files in Stata 15 format (118)
+
+    DTA 118 format files support unicode string data (both fixed and strL)
+    format. Unicode is also supported in value labels, variable labels and
+    the dataset label.
+
+    .. versionadded:: 0.26.0
```

Comment:
1.0

Reply:
Fixed.  I went with 1.0.0.  Is this right, or just 1.0?

Reply:
i think we've been using 1.0.0